spec.go

```diff
@@ -6,20 +6,55 @@ package spec
 
 import (
 	"embed"
+	"fmt"
 	"io/fs"
+
+	"github.com/Masterminds/semver/v3"
+	"gopkg.in/yaml.v3"
 )
 
-//go:embed versions/1 versions/1/integration/_dev versions/1/integration/data_stream/_dev versions/1/input
+//go:embed spec spec/integration/_dev spec/integration/data_stream/_dev spec/input
 var content embed.FS
 
-// FS returns an io/fs.FS for accessing the "package-spec/version" contents.
-//
-// All contents are rooted within the "versions" folder, and are addressed first by
-// version, e.g. "1/spec.yml".
+// FS returns an io/fs.FS for accessing the "package-spec/spec" contents.
 func FS() fs.FS {
-	fs, err := fs.Sub(content, "versions")
+	fs, err := fs.Sub(content, "spec")
 	if err != nil {
 		panic(err)
 	}
 	return fs
 }
+
+// CheckVersion checks if the given version is implemented by current spec. It returns
+// the version of the spec matching with the given version.
+func CheckVersion(version *semver.Version) (*semver.Version, error) {
+	d, err := fs.ReadFile(content, "spec/changelog.yml")
+	if err != nil {
+		panic(err)
```

Comment:
Is it safe to panic here? Is it executed only during `init`?

Reply:
I followed the same approach as in `FS()` when accessing the filesystem. But yeah, it'd be better to don't panic here.